SwiftWisdom/Core/StandardLibrary/Numbers/Int+Extensions.swift

```diff
@@ -24,3 +24,34 @@ extension IntegerType {
         (0..<self).forEach { _ in closure() }
     }
 }
+
+extension IntegerType {
+    public func ip_toMagnitudeString(decimalPlaces decimalPlaces: Int = 1) -> String {
+        
+        guard self > 999 else { return "\(self)" }
+        
+        let units = ["K", "M", "B", "T", "Q"]
+        
+        let value = Double(self.toIntMax())
+        
+        var magnitude: Int = Int(log10(value) / 3.0) // the order of magnitude of our like count in thousands
+        
+        // divide value by 1000^magnitude to get hundreds value, then round to desired decimal places
+        var roundedHundredsValue = (value / pow(1000.0, Double(magnitude))).ip_roundToDecimalPlaces(decimalPlaces)
+        
+        // if rounding brings our display value over 1000, divide by 1000 and then bump the magnitude
+        if roundedHundredsValue >= 1000 {
+            roundedHundredsValue /= 1000.0
+            magnitude += 1
+        }
+        
+        let formatter = DecimalFormatter.sharedInstance
+        guard let likesFormatted = formatter.stringFromNumber(roundedHundredsValue) else {
+            return "\(roundedHundredsValue)\(units[magnitude-1])"
```

Comment:
I'm still concerned here about crash potential. I know it's rare, but seems like we should check that this index is within range first and offer some type of alternative to crash.